SignalServiceKit/src/Storage/FullTextSearchFinder.swift

```diff
@@ -47,27 +62,31 @@ public class FullTextSearchFinder: NSObject {
         }
     }
 
-    private func ext(transaction: YapDatabaseReadTransaction) -> YapDatabaseFullTextSearchTransaction? {
-        return transaction.ext(FullTextSearchFinder.dbExtensionName) as? YapDatabaseFullTextSearchTransaction
-    }
-
-    // Mark: Index Building
+    private func normalize(queryText: String) -> String {
+        var normalized: String = queryText.trimmingCharacters(in: .whitespacesAndNewlines)
 
-    private class var contactsManager: ContactsManagerProtocol {
-        return TextSecureKitEnv.shared().contactsManager
-    }
-
-    private class func normalize(text: String) -> String {
-        var normalized: String = text.trimmingCharacters(in: .whitespacesAndNewlines)
-
-        // Remove any phone number formatting from the search terms
+        // Remove any punctuation from the search terms
         let nonformattingScalars = normalized.unicodeScalars.lazy.filter {
             !CharacterSet.punctuationCharacters.contains($0)
         }
+        let normalizedChars = String(String.UnicodeScalarView(nonformattingScalars))
 
-        normalized = String(String.UnicodeScalarView(nonformattingScalars))
+        let digitsOnlyScalars = normalized.unicodeScalars.lazy.filter {
+            CharacterSet.decimalDigits.contains($0)
+        }
+        let normalizedDigits = String(String.UnicodeScalarView(digitsOnlyScalars))
 
-        return normalized
+        if normalizedDigits.count > 0 {
```

Comment:
So, a query like: "(321) 321-4321" would match a message body like "321 salesmen sold 321 peanuts to 4321 elephants"?  That sounds fine.

Reply:
No, note there are now two different normalization methods, one for the query text the user types into the search bar and one which the yap extension uses when building the search index.

The digit normalization only happens on the query side.

The indexing side makes sure that both the e164 and the national number are stored in the index for any recipient id.

So the inverse is true, a query of "321 salesmen sold 321 peanuts to 4321 elephants" would match a recipient with id "+13213214321"